app/models/classification.rb

```diff
@@ -3,27 +3,29 @@ class Classification < ApplicationRecord
 
   belongs_to :tag
 
+  virtual_column :name, :type => :string
+
   before_save    :save_tag
   before_destroy :delete_tags_and_entries
-  validate       :validate_format_of_name, :validate_uniqueness_on_tag_name
-
-  validates_uniqueness_of :description, :scope => [:parent_id],
-                                        :if    => proc { |c|
-                                          cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
-                                          unless c.new_record?
-                                            cond.first << " AND id != ?"
-                                            cond << c.id
-                                          end
-                                          c.class.in_region(region_id).exists?(cond)
-                                        }
-  validates_presence_of :name, :description
-  validates_length_of :name, :maximum => 50, :message => "must not exceed 50 characters"
-  validates_length_of :description, :maximum => 255, :message => "must not exceed 255 characters"
-  validates_inclusion_of :syntax,
-                         :in      => %w( string integer boolean ),
-                         :message => "should be one of 'string', 'integer' or 'boolean'"
 
-  virtual_column :name, :type => :string
+  validates :description, :presence => true, :length => {:maximum => 255}
+  validates :description, :uniqueness => {:scope => [:parent_id]}, :if => proc { |c|
+    cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
+    unless c.new_record?
+      cond.first << " AND id != ?"
+      cond << c.id
+    end
+    c.class.in_region(region_id).exists?(cond)
+  }
```

Comment:
ok, sorry - you changed indentation on this code.

@chessbyte yes, this can be done better with arel. and it should probably just be a 100% custom validator since the `scope` is not expressive enough for us.

Think fixing this validator is out of scope for this PR